paper-button-state.html

```diff
@@ -0,0 +1,188 @@
+<!--
+Copyright (c) 2015 The Polymer Project Authors. All rights reserved.
+This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE
+The complete set of authors may be found at http://polymer.github.io/AUTHORS
+The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS
+Code distributed by Google as part of the polymer project is also
+subject to an additional IP rights grant found at http://polymer.github.io/PATENTS
+-->
+
+<link rel="import" href="../polymer/polymer.html">
+<script>
+  (function() {
+    var FocusSource = {
+      MOUSE: 'mouse',
+      KEYBOARD: 'keyboard',
+      NONE: 'none'
+    };
+
+    Polymer({
+      is: 'paper-button-state',
+
+      properties: {
+        /**
+         * The elevation of the button, represented as 2dp. Note that this
+         * is a computed property and should be treated as read-only.
+         *
+         * @attribute elevation
+         * @type boolean
+         * @default 0
+         */
+        elevation: {
+          type: Number,
+          notify: true,
+          computed: 'computeElevation(focused, focusSource, active, disabled, raised)'
+        },
+
+        /**
+         * If true, the button will use toggle conventions for activation and
+         * deactivation.
+         *
+         * @attribute toggle
+         * @type boolean
+         * @default false
+         */
+        toggle: {
+          type: Boolean,
+          readOnly: true,
+          notify: true,
+          value: false
+        },
+
+        /**
+         * If true, the button should be styled with a shadow.
+         *
+         * @attribute raised
+         * @type boolean
+         * @default false
+         */
+        raised: {
+          type: Boolean,
+          value: false,
+          notify: true,
+          observer: 'onRaisedChanged'
+        },
+
+        focusSource: {
+          type: String,
+          readOnly: true,
+          notify: true,
+          value: function() {
+            this._setFocusSource(FocusSource.NONE);
+          }
+        },
+
+        active: {
+          type: Boolean,
+          value: false,
+          observer: 'onActiveChanged'
+        },
+
+        focused: {
+          type: Boolean,
+          value: false,
+          observer: 'onFocusedChanged'
+        },
+
+        disabled: {
+          type: Boolean,
+          value: false
+        }
+      },
+
+      get target() {
+        return this.host || this.parentNode;
+      },
+
+      configure: function() {
+        return {
+          _focusSource: FocusSource.NONE,
+          _attributeObserver: new MutationObserver(
+            this.onTargetAttributesChanged.bind(this)
+          )
+        };
+      },
+
+      ready: function() {
+        if (!this.target.hasAttribute('role')) {
+          this.target.setAttribute('role', 'button');
```

Comment:
The mixin doesn't own the DOM of the host any more than the composed element does. At least the composed element naturally knows about its host or parent via relative DOM references. A mixin, by contrast, must assume that it is being mixed into something that happens to be a DOM element, and does not receive a guarantee from the platform that `this` is an element.